nav2_util/src/node_utils.cpp

```diff
@@ -63,13 +63,10 @@ std::string generate_internal_node_name(const std::string & prefix)
   return sanitize_node_name(prefix) + "_" + time_to_string(8);
 }
 
-rclcpp::Node::SharedPtr generate_internal_node(const std::string & prefix)
-{
-  rclcpp::NodeOptions options;
-  options.use_global_arguments(false);
-  options.start_parameter_services(false);
-  options.start_parameter_event_publisher(false);
-  return rclcpp::Node::make_shared(generate_internal_node_name(prefix));
+rclcpp::Node::SharedPtr generate_internal_node(const std::string& prefix) {
+  auto options = rclcpp::NodeOptions().arguments(
+      {"__node:=" + generate_internal_node_name(prefix)});
+  return rclcpp::Node::make_shared("_", options);
```

Comment:
How about bringing across the start_parameter_services(false) and start_parameter_event_publisher(false).  I assume these both default to true. We're generally trying to cut down on the number of topics/services, etc. if they aren't needed.

Reply:
Sure. I don't know the implications of that change but you got it!